src/software/new_geom/segment.h

```diff
@@ -9,27 +9,43 @@
 // TODO (Issue #1098): Refactor this to fit the new_geom hierarchy
 class Segment final
 {
-   private:
-    Point start;
-    Point end;
-
    public:
+    /**
+     * Sets the start point of the segment to o
+     *
+     * @param o new start point of segment
+     */
     void setSegStart(Point o)
     {
         start = o;
     }
 
-    Point getSegStart() const
+    /**
+     * Gets the start point of the segment
+     *
+     * @return start point of segment
+     */
+    const Point& getSegStart() const
     {
         return start;
     }
 
+    /**
+     * Sets the end point of the segment to o
+     *
+     * @param o new end point of segment
+     */
```

Comment:
Mismatch of 'o' in comment and 'f' in function

Reply:
fixed